### DISCUSSION TITLE
404 Error when accessing Pendant UI v1 API endpoints, except for the main API page

### DISCUSSION THREAD

**Discussion by Turner66** [winder/Universal-G-Code-Sender#2250] [Help] [answered]
Hello,

I am trying to use the Pendant UI v1 API to retrieve the current machine status, but I am receiving a 404 error when attempting to access the `/api/v1/machine/status` endpoint. I have confirmed that the Pendant UI API is running on the specified server and port, and I am able to access the main API page, but all other endpoints, such as the machine status endpoint, are returning a 404 error. I am unable to find any documentation for this specific API. Can you please provide some guidance on how to use this API and resolve this issue?

Thank you.

**Comment by chrisla23** [comment, marked as answer]
Are you using getStatus?

Here is a working example I am using in a batch script:

FOR /F "delims=" %%A IN ('xidel.exe
http://192.168.7.121:8080/api/v1/status/getStatus -e
"sendDuration:=($json).sendDuration" -e
"sendRemainingDuration:=($json).sendRemainingDuration"  -e
"workCoordX:=round(($json).workCoord.x,3)" -e
"workCoordY:=round(($json).workCoord.y,3)" -e
"workCoordZ:=round(($json).workCoord.z,3)" -e "state:=($json).state"
--output-format=cmd') DO %%A


Thanks,



-Chris

**Reply by breiler** [reply]
To expand on the answer from @chrisla23. There is unfortunately no API documentation. But you should be able to figure out the available API endpoints by browsing the source code here: 

https://github.com/winder/Universal-G-Code-Sender/tree/master/ugs-pendant/src/main/java/com/willwinder/universalgcodesender/pendantui/v1/resources

**Reply by Turner66** [reply]
@chrisla23 Thank you very much. Do you have a collection of API calls that you could share. I would save a lot of time and nerves.Love greetings.

**Reply by Turner66** [reply]
@breiler Can you explain how the file transfer process over WiFi works and if it is possible to implement this functionality in a custom app ?

**Comment by chrisla23** [comment]
I don't but I originally trolled through this code, which serves as a
pretty clean example (starting at about line 143):

https://github.com/aleslukek/UGS-Wifi-Pendant/blob/master/UGS_Wifi_Pendant/d_presetup.ino

Thanks,

-Chris

**Reply by Turner66** [reply]
Can you explain how the file transfer process over WiFi works and if it is possible to implement this functionality in a custom app ?
